### PR TITLE
DBZ-673 txId must be long in wal2jon decoder

### PR DESCRIPTION
**Pull request by glistman**
the getInteger method returns a null when the value txid is not integer
37851938-b7568820-2eae-11e8-8760-3701f1314e97

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public void processMessage(ByteBuffer buffer, ReplicationMessageProcessor proces
             final byte[] content = Arrays.copyOfRange(source, buffer.arrayOffset(), source.length);
             final Document message = DocumentReader.floatNumbersAsTextReader().read(content);
             LOGGER.debug("Message arrived for decoding {}", message);
-            final int txId = message.getInteger("xid");
+            final long txId = message.getLong("xid");
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java]: Can you do the equivalent adjustments to `StreamingWal2JsonMessageDecoder`, too?

**Comment by gunnarmorling** [comment]
Hey @jpechane, just seeing that it's handled as `int` in `PgProto.RowMessage.getTransactionId()`, too. So that'd also need adjustment. I suppose this also implies some changes to the logical decoding plug-in are needed?

**Comment by gunnarmorling** [comment]
Ok, so it's `uint32` in the ProtoBuf schema already (which I think is enough). But as per the ProtoBuf docs and [this post](https://stackoverflow.com/questions/14990261/how-do-i-convert-mysql-unsigned-ints-to-protocol-buffer-uint32s-in-java) it keeps the highest bit in the sign bit of the `int`, so some special care is needed when converting this value to `long`.

**Comment by jpechane** [comment]
I see the problem - tx id is `uint32` in `C++` but `int32` in protobuf. According to the docs this is fine if you mostly send non-negative values. We should switch to `uint32` too and at the same time modify our Java code to handle large integers
> In Java, unsigned 32-bit and 64-bit integers are represented using their signed counterparts, with the top bit simply being stored in the sign bit.

**Comment by gunnarmorling** [comment]
Exactly. So essentially `PgProto.RowMessage.getTransactionId()` must never be called by directly but rather should go threw some helper method which deals with the correct conversion into `long` also for values > 2^31.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java (hunk 1 above)
Comment (glistman) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java]: yes of course

**Comment by glistman** [comment]
I use Integer.toUnsignedLong to wrap PgProto.RowMessage.getTransactionId () before returning the txid

**Comment by gunnarmorling** [comment]
Excellent, wasn't even aware of `toUnsignedLong()`. I added one more commit to apply this conversion in one more place and merged it. Thanks, @glistman!

**Pull request merged**